presto-orc/src/main/java/com/facebook/presto/orc/OrcReader.java

```diff
@@ -490,4 +492,23 @@ public OrcDataSource getOrcDataSource()
     {
         return orcDataSource;
     }
+
+    public StripeFooter readStripeFooter(StripeInformation stripe)
+            throws IOException
+    {
+        requireNonNull(stripe, "stripe is null");
+
+        byte[] tailBuffer = new byte[toIntExact(stripe.getFooterLength())];
+        orcDataSource.readFully(stripe.getOffset() + stripe.getIndexLength() + stripe.getDataLength(), tailBuffer);
+        try (InputStream inputStream = new OrcInputStream(
+                orcDataSource.getId(),
+                new SharedBuffer(NOOP_ORC_LOCAL_MEMORY_CONTEXT),
```

Comment:
I used noop aggregating and local memory contexts because the memory footprint is small and non-persistent.